src/ardrive.ts

```diff
@@ -77,6 +116,57 @@ export class ArDriveAnonymous extends ArDriveType {
 		const driveEntity = await this.arFsDao.getPublicDrive(driveId);
 		return Promise.resolve(driveEntity);
 	}
+
+	async getPublicFolder(folderId: string): Promise<ArFSPublicFolder> {
+		const folder = await this.arFsDao.getPublicFolder(folderId);
+		return folder;
+	}
+
+	async getChildrenOfPublicFolder(folderId: FolderID): Promise<ArFSPublicFileOrFolderData[]> {
+		const folder = await this.arFsDao.getPublicFolder(folderId);
+
+		// Fetch all of the folder entities within the drive
+		const driveIdOfFolder = folder.driveId;
+		const allFolderEntitiesOfDrive = (await this.arFsDao.getAllFoldersOfPublicDrive(driveIdOfFolder)).filter(
+			lastFolderRevisionFilter
+		);
+
+		// Feed entities to FolderHierarchy.setupNodesWithEntity()
+		const hierarchy = FolderHierarchy.newFromEntities(allFolderEntitiesOfDrive);
+		const folderIDs = hierarchy.allFolderIDs();
+
+		// Fetch all file entities within all Folders of the drive
+		const allFileEntitiesOfDrive = (await this.arFsDao.getAllPublicChildrenFilesFromFolderIDs(folderIDs)).filter(
+			lastFileRevisionFilter
+		);
+
+		const allEntitiesOfDrive = [...allFolderEntitiesOfDrive, ...allFileEntitiesOfDrive];
+		const childrenFolderIDs = hierarchy.subTreeOf(folderId).allFolderIDs();
+		const allChildrenOfFolder = allEntitiesOfDrive.filter(childrenAndFolderOfFilterFactory(childrenFolderIDs));
+
+		const mergedData = allChildrenOfFolder.map((entity) => {
+			const path = `${hierarchy.pathToFolderId(entity.parentFolderId)}${entity.name}`;
+			const txPath = `${hierarchy.txPathToFolderId(entity.parentFolderId)}${entity.txId}`;
+			const entityIdPath = `${hierarchy.entityPathToFolderId(entity.parentFolderId)}${entity.entityId}`;
+			return new ArFSPublicFileOrFolderData(
```

Comment:
How about a static factory function that takes an entity and the paths and does all this annoying unpacking internally?

Reply:
Now this is handled in the `WithPaths` entities constructor
a7d657d715e552dc6905c61eaa0593fc759166e3